test/migrations/20200401161126_move_nmbas_to_project_level.js

```diff
@@ -0,0 +1,448 @@
+          protocols: [
+            {
+              title: 'protocol 1',
+              steps: [
+                {
+                  nmbas: true
+                }
+              ]
+            }
+          ]
+        }
+      },
+      {
+        project_id: ids.active,
+        status: 'draft',
+        data: {
+          title: 'Active Project',
+          protocols: [
+            {
+              title: 'protocol 1',
+              steps: [
+                {
+                  nmbas: true
+                }
+              ]
+            }
+          ]
+        }
+      },
+      {
+        project_id: ids.draft,
+        status: 'submitted',
+        data: {
+          title: 'Draft Project',
+          protocols: [
+            {
+              title: 'protocol 1',
+              steps: [
+                {
+                  nmbas: true
+                }
+              ]
+            }
+          ]
+        }
+      },
+      {
+        project_id: ids.draft,
+        status: 'draft',
+        data: {
+          title: 'Draft Project',
+          protocols: [
+            {
+              title: 'protocol 1',
+              steps: [
+                {
+                  nmbas: true
+                }
+              ]
+            }
+          ]
+        }
+      },
+      {
+        project_id: ids.noNmbas,
+        status: 'granted',
+        data: {
+          title: 'Project no NMBAs',
+          protocols: [
+            {
+              title: 'protocol 1',
+              steps: [
+                {
+                  nmbas: false
+                }
+              ]
+            }
+          ]
+        }
+      }
+    ];
+
+    before(() => {
+      this.knex = db.init();
+    });
+
+    beforeEach(() => {
+      return Promise.resolve()
+        .then(() => db.clean(this.knex))
+        .then(() => this.knex('establishments').insert(establishment))
+        .then(() => this.knex('profiles').insert(licenceHolder))
+        .then(() => this.knex('projects').insert(projects))
+        .then(() => this.knex('project_versions').insert(versions));
+    });
+
+    afterEach(() => {
+      return db.clean(this.knex);
+    });
+
+    after(() => {
+      return this.knex.destroy();
+    });
+
+    it('is ok', () => {
+      assert.ok(true);
+    });
+
+    it('does not change any properties, only adds nmbas-used', () => {
+      return Promise.resolve()
+        .then(() => this.knex('project_versions').select('project_versions.*').orderBy('id'))
+        .then(before => {
+          return Promise.resolve()
+            .then(() => {
+              return up(this.knex);
+            })
+            .then(() => this.knex('project_versions').select('project_versions.*').orderBy('id'))
+            .then(after => {
+              const changes = diff(before, after);
+              assert.ok(changes.every(change => {
+                return change.kind === 'N' && change.path.pop() === 'nmbas-used';
+              }));
```

Comment:
I think for this one, because it's a single top-level property the path should exactly equal `['nmbas-used']`

Also the diff should only contain a single entry.

Reply:
`changes` looks like this:
```[ DiffNew { kind: 'N', path: [ 6, 'data', 'nmbas-used' ], rhs: true },
  DiffNew { kind: 'N', path: [ 5, 'data', 'nmbas-used' ], rhs: true },
  DiffNew { kind: 'N', path: [ 2, 'data', 'nmbas-used' ], rhs: true },
  DiffNew { kind: 'N', path: [ 1, 'data', 'nmbas-used' ], rhs: true },
  DiffNew { kind: 'N', path: [ 0, 'data', 'nmbas-used' ], rhs: true } ]```

Reply:
Yes, I missed that you were passing in an array of _all_ versions and not a specific version's data.

Carry on.